### ISSUE TITLE
Undefined method enum_slice

### ISSUE THREAD

**Issue by carlinisaacson** [JulienLeonard/XRVG#1]
I am following the tutorial and when executing this code:

``` ruby
require 'xrvg'
include XRVG

render = SVGRender[ :filename, "hellocrown2.svg" ]
SyncS[Circle[], (0.2..0.1)].samples( 10 ) do |point, radius|
  render.add( Circle[:center, point, :radius, radius ], Style[ :fill, Color.blue( 0.5 ) ] )
end
render.end
```

I get this error.

``` ruby
/Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/utils.rb:390:in `foreach': undefined method `enum_slice' for #<Array:0x007ffcbb8d2380> (NoMethodError)
    from /Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/utils.rb:436:in `block in forzip'
    from /Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/utils.rb:435:in `zip'
    from /Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/utils.rb:435:in `forzip'
    from /Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/samplation.rb:322:in `compute'
    from /Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/samplation.rb:64:in `process'
    from /Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/samplation.rb:51:in `trigger'
    from /Users/cmi/.gem/ruby/2.1.5/gems/xrvg-0.0.82/lib/samplation.rb:193:in `samples'
    from rvg.rb:5:in `<main>'
```

From what I can find it appears using Floats within a range is the issue, but the documentation says that support for Floats within ranges was built into XRVG. So, am I looking in the wrong direction for a solution?

**Comment by JulienLeonard** [comment]
Hi, may I know which Ruby version you are using ? It looks like the enum_slice method is deprecated in ruby 1.9....

**Comment by carlinisaacson** [comment]
I am using ruby 2.1.5. The oldest ruby version I have installed on my computer is 1.9.3 and I could probably delete it because I don't ever use it.

**Comment by JulienLeonard** [comment]
Ok, enum_slice has been replaced by each_slice from ruby 1.9 onwards ...

**Comment by JulienLeonard** [comment]
I am afraid I cannot update for the moment the source and test it ... Could you try it ?

**Comment by mhintz** [comment]
I tried just replacing the use of `enum_slice` with `each_slice` and it works for me on Ruby 2.2.1

**Comment by JulienLeonard** [comment]
Hi Mark,

Thanks for your update.
Need to update my gem then...

Regards
Julien

**Comment by mhintz** [comment]
In addition, on line 404 of utils.rb, replacing enum_cons with each_cons seems to work. (Otherwise an error is thrown under 2.2.1)

**Comment by JulienLeonard** [comment]
Hi Mark,

OK noted. Thanks a lot.

2015-04-06 2:13 GMT+10:00 Mark Hintz notifications@github.com:

**Comment by JulienLeonard** [comment]
Issue patched and corrected with gem version 0.0.85. Thanks all for the notice and the checking.
